### ISSUE TITLE
bug: Cursor position doesn't auto align on directory enter when ../ (parent directory) item is shown

### ISSUE THREAD

**Issue by johnpgr** [stevearc/oil.nvim#536]
### Did you check the docs and existing issues?

- [X] I have read the docs
- [X] I have searched the existing issues

### Neovim version (nvim -v)

v0.10.2

### Operating system/version

Ubuntu 24.04.1 LTS

### Describe the bug

When first entering a directory in Oil, the cursor gets automatically positioned to the first file name start.
That doesn't happen if the hidden files are enabled, because the first file is now `../` (parent directory)

### How it behaves with hidden files disabled

https://github.com/user-attachments/assets/7aa67544-3365-4847-bfe2-709ed6f85bc1

### How it behaves with hidden files enabled

https://github.com/user-attachments/assets/018fd0c1-4414-4da5-b7f4-f694407a1c56

### What is the severity of this bug?

minor (annoyance)

### Steps To Reproduce

1. toggle hidden files
2. navigate between directories

### Expected Behavior

The cursor position on first directory enter should have a consistent behavior with hidden files enabled.

### Directory structure

any directory structure

### Repro

```Lua
-- save as repro.lua
-- run with nvim -u repro.lua
-- DO NOT change the paths
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "runtime", "cache" }) do
	vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
	vim.fn.system({
		"git",
		"clone",
		"--filter=blob:none",
		"--single-branch",
		"https://github.com/folke/lazy.nvim.git",
		lazypath,
	})
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
	"folke/tokyonight.nvim",
	{
		"stevearc/oil.nvim",
		config = function()
			require("oil").setup({
				columns = {
					"permissions",
					"size",
					"ctime",
					"icon",
				},
			})
		end,
	},
	-- add any other plugins here
}
require("lazy").setup(plugins, {
	root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here
```

### Did you check the bug with a clean config?

- [X] I have confirmed that the bug reproduces with `nvim -u repro.lua` using the repro.lua file above.

**Comment by BozeBro** [comment]
I cannot seem to reproduce based off of your repro.lua file. I do not 

With hidden files off
<img width="1093" alt="Screenshot 2025-01-07 at 7 55 55 PM" src="https://github.com/user-attachments/assets/71460186-e7d2-4323-bc36-a3b4adc0a1fd" />

With hidden files on

<img width="1118" alt="Screenshot 2025-01-07 at 7 56 14 PM" src="https://github.com/user-attachments/assets/6dae8c81-8d38-40e6-93b9-053e0b86719c" />

I tried on both Ubuntu and macOS

**Comment by johnpgr** [comment]
Yes, I apologize, I did forgot to add the options that I use, that causes this bug. I edited the `repro.lua` config.
The bug should now be reproduceable.
It's due to me using the plugin for so much time with the additional columns now that I forgot that they are not a default in oil.

**Comment by BozeBro** [comment]
A workaround for this is if you set this in your config `constrain_cursor = "name"`
This will put your cursor in the expected place. I'm still looking around at this line https://github.com/stevearc/oil.nvim/blob/09fa1d22f5edf0730824d2b222d726c8c81bbdc9/lua/oil/view.lua#L279 which seems to returning line information that is not fully consistent for when the file name is `../`

**Comment by BozeBro** [comment]
Made a PR, that should hopefully fix the issue fully.
@johnpgr This should hopefully solve your problems without the workaround I mentioned earlier.

When entering a directory, since .. is not stored in the entry cache, oil will not change the position of the cursor in the line
Idea here is to check if the id we're looking at is the `..` directory. 
This should be ok because `..` is special, and it seems like .. is always set to id 0 anyway. 
@stevearc feel free to lmk if that assumption is incorrect.

[PR](https://github.com/stevearc/oil.nvim/pull/562)

**Comment by stevearc** [comment]
Should be fixed now
